src/seq/index.rs

```diff
@@ -280,44 +280,72 @@ where
     F: Fn(usize) -> X,
     X: Into<f64>,
 {
-    if amount == 0 {
-        return Ok(IndexVec::USize(Vec::new()));
+    if length > (::core::u32::MAX as usize) {
+        sample_efraimidis_spirakis(rng, length, weight, amount)
+    } else {
+        let amount = amount as u32;
+        let length = length as u32;
+        sample_efraimidis_spirakis(rng, length, weight, amount)
+    }
```

Comment:
Should we care about the corner case where `length <= u32::MAX` but `amount` is not?

Since `sample_efraimidis_spirakis` is an internal method, we could just move the sanity checks on input up to this method.

Reply:
I see there's now an assert on this path; that's good enough.